sql/core/src/main/scala/org/apache/spark/sql/functions.scala

```diff
@@ -3488,6 +3511,16 @@ object functions {
    * Applies a binary operator to an initial state and all elements in the array,
    * and reduces this to a single state. The final state is converted into the final result
    * by applying a finish function.
+   * {{{
+   *   df.select(aggregate(col("i"), lit(0), (acc, x) => acc + x, _ * 10))
+   * }}}
+   *
+   * @param expr the input array column
+   * @param zero the initial value
```

Comment:
Since this are polishing this part, can we rename this column from `zero` to a more meaningful name (e.g. `initialValue`)?

Reply:
Hm, does that change the API? I forget when param names matter in Scala.
In any event, 'zero' isn't a terrible name as a reduce-like function needs a 'zero' value that is the result of no aggregation.

Reply:
Sometimes, it's `1` instead of zero. For example, `product` instead of `summation`?

Reply:
Good point. The scala API calls it `z` in `fold` for example. Well if it isn't flagged as an API change by MiMa, OK by me.

Reply:
Thanks. Got it. Since this is a new API added at 3.0.0, `MiMa` will be okay.

Reply:
Thanks all! I test Mima locally and it's good.